src/Purebred/Parsing/Text.hs

```diff
@@ -31,3 +41,27 @@ niceEndOfInput = endOfInput <|> p
 -- | Get the current position of the parser
 offset :: AT.Parser i Int
 offset = AT.Parser $ \t pos more _lose suc -> suc t pos more (AT.fromPos pos)
+
+parseMailbody :: Source -> T.Text -> MailBody
+parseMailbody s =
+  either
+    (\e -> MailBody s [Paragraph [Line [] 0 (T.pack e)]])
+    (MailBody s . setLineNumbers) . parseOnly (paragraphs <* niceEndOfInput)
+
+endOfParagraph :: Parser ()
+endOfParagraph = endOfLine *> endOfLine
+
+paragraph :: Parser Paragraph
+paragraph = Paragraph . makeLines . T.pack <$> manyTill anyChar endOfParagraph
+
+paragraphs :: Parser [Paragraph]
+paragraphs = do
+  paras <- many' paragraph
+  rest <- takeText
+  pure $ paras <> [Paragraph $ makeLines rest]
+
+makeLines :: T.Text -> [Line]
+makeLines = fmap (Line [] 0) . wrapTextToLines defaultWrapSettings 82
```

Comment:
Should make the width perhaps configurable unless I find a way to add an additional custom widget.